React InstantSearch/routing-basic/src/App.js

```diff
@@ -25,70 +25,55 @@ const searchStateToUrl = (props, searchState) =>
 
 const urlToSearchState = location => qs.parse(location.search.slice(1));
 
-class App extends Component {
-  state = {
-    searchState: urlToSearchState(this.props.location),
-    lastLocation: this.props.location,
-  };
+export function App(props) {
+  const { location, history } = props;
+  const [searchState, setSearchState] = useState(urlToSearchState(location));
+  const [debouncedSetState, setDebouncedSetState] = useState(null);
```

Comment:
This is not a state variable because we don't want to re-render when it changes. We can use a ref here.

Reply:
Addressed in 5e9f823.